Foundation/String.swift

```diff
@@ -1579,43 +1579,26 @@ extension String {
             ) != nil
         return r
     }
-    
-#if os(Linux)
+}
+
+#if !_runtime(_ObjC)
+import CoreFoundation
+
+extension String {
     public func hasPrefix(prefix: String) -> Bool {
-        let characters = utf16
-        let prefixCharacters = prefix.utf16
-        let start = characters.startIndex
-        let prefixStart = prefixCharacters.startIndex
-        if characters.count < prefixCharacters.count {
-            return false
-        }
-        for idx in 0..<prefixCharacters.count {
-            if characters[start.advancedBy(idx)] != prefixCharacters[prefixStart.advancedBy(idx)] {
-                return false
-            }
-        }
-        return true
+        let cfstring = self._cfObject
+        let range = CFRangeMake(0, CFStringGetLength(cfstring))
+        let opts = CFStringCompareFlags(kCFCompareAnchored + kCFCompareNonliteral)
```

Comment:
+? shouldn't that be an |?
This might be cleaner just to funnel to the NS version of 
rangeOfString(str, options: .AnchoredSearch, range: NSMakeRange(0, length)).location != NSNotFound